sdk/python/feast/infra/gcp.py

```diff
@@ -101,11 +102,12 @@ def online_write_batch(
     ) -> None:
         client = self._initialize_client()
 
-        pool = ThreadPool(processes=40)
-        pool.map(
-            lambda b: _write_minibatch(client, project, table, b, progress),
-            _to_minibatches(data),
-        )
+        with tqdm(total=len(data)) as pbar:
```

Comment:
I think it may be better to have tqdm at a higher level so that its not optional to do logging. Would it make more sense in `materialize_incremental()`?

Reply:
I don't see an easy way to bring it up to materialize_incremental since we don't know the number of rows of data at that level; however, `progress` is an argument in methods in the `Provider` class so that kinda makes it not optional for implementations of `Provider` to do some sort of progress logging, presumably by copying existing implementations and using `tqdm`

Reply:
I thought I replied here, but somehow my message didn't send. I think my question is just: How can we ensure consistency in the logging output if the implementations need to create their own tqdm object. Wouldn't the current implementation make it unscalable to have different configurations for tqdm? Even if the row count is deeper, maybe it makes sense to instantiate the tqdm object higher and pass it into the implementation.